packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,642 @@
+        transactionCommittedDetailsResponse.transaction.confirmed_at !== undefined &&
+        transactionCommittedDetailsResponse.transaction.intent_hash !== undefined
+      ) {
+        const transaction: Tx = await this.convertTransactionFromHex(
+          transactionCommittedDetailsResponse.transaction.raw_hex,
+          transactionCommittedDetailsResponse.transaction.intent_hash,
+          transactionCommittedDetailsResponse.transaction.confirmed_at,
+        )
+        return transaction
+      } else {
+        throw new Error('Incomplete transaction data received')
+      }
+    } catch (error) {
+      throw new Error('Failed to fetch transaction data')
+    }
+  }
+
+  /**
+   * Helper function to convert a transaction in hex, returned by the gateway to a Tx type
+   * @param transaction_hex - The raw_hex returned by the gateway for a transaction id
+   * @param confirmed_at - The confirmed_at date for the transaction
+   * @param intent_hash - The transaction intent hash
+   * @returns a transaction in Tx type
+   */
+  async convertTransactionFromHex(transaction_hex: string, intent_hash: string, confirmed_at: Date): Promise<Tx> {
+    const transactionBinary = Convert.HexString.toUint8Array(transaction_hex)
+    try {
+      const transactionSummary = await LTSRadixEngineToolkit.Transaction.summarizeTransaction(transactionBinary)
+      const withdrawAccount = Object.keys(transactionSummary.withdraws)[0]
+      const depositAccount = Object.keys(transactionSummary.deposits)[0]
+      const withdrawResource = Object.keys(transactionSummary.withdraws[withdrawAccount])[0]
+      const withdrawAmount: number = transactionSummary.withdraws[withdrawAccount][withdrawResource].toNumber()
+
+      const transaction: Tx = {
+        from: [
+          {
+            from: withdrawAccount,
+            amount: baseAmount(withdrawAmount),
+            asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: RadixChain },
+          },
+        ],
+        to: [
+          {
+            to: depositAccount,
+            amount: baseAmount(withdrawAmount),
+            asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: RadixChain },
+          },
+        ],
+        date: confirmed_at,
+        type: TxType.Transfer,
+        hash: intent_hash,
+        asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: RadixChain },
+      }
+      return transaction
+    } catch (error) {
+      return {
+        from: [],
+        to: [],
+        asset: XrdAsset,
+        date: confirmed_at,
+        type: TxType.Unknown,
+        hash: intent_hash,
+      }
+    }
+  }
+
+  /**
+   * Creates a transaction using the SimpleTransactionBuilder, signs it with the
+   * private key and returns the signed hex
+   * @param params - The transactions params
+   * @returns A signed transaction hex
+   */
+  async transfer(params: TxParams, broadcastTx = true): Promise<string> {
```

Comment:
We can't have a `broadcastTx` here since this would not be compliant with the XChainJs required interface. We can just remove this argument and have this function construct and submit to the network which is what other clients do.